src/goal/java/adoptiumJava.ts

```diff
@@ -0,0 +1,110 @@
+import { setIfAbsent } from "#common/general.ts";
+import { defineGoal, type VersionOutput } from "#core/goal.ts";
+import adoptiumJavaVersions from "#provider/java/adoptiumJavaVersions.ts";
+import type { VersionFileRuntime } from "#schema/format/v1/versionFile.ts";
+import type { AdoptiumJavaBinary, AdoptiumJavaRuntimeEntries, AdoptiumJavaRuntimeEntry } from "#schema/java/adoptiumJavaData.ts";
+import { orderBy } from "es-toolkit";
+
+export default defineGoal({
+	id: "net.adoptium.java",
+	name: "Adoptium Provided Java",
+	provider: adoptiumJavaVersions,
+
+	generate(info): VersionOutput[] {
+		const result: VersionOutput[] = [];
+
+		const majorVersions: Map<number, AdoptiumJavaRuntimeEntries> = new Map;
+
+		for (const entry of info) {
+			if (entry.binaries.some(x => isAvailableBinary(x))) {
+				setIfAbsent(majorVersions, entry.version_data.major, []).push(entry);
+			}
+		}
+
+		for (const [majorVersion, entries] of majorVersions) {
+			majorVersions.set(
+				majorVersion,
+				orderBy(entries, [entry => entry.timestamp], ["desc"])
+			);
+		}
+
+		for (const [majorVersion, entries] of majorVersions) {
+			result.push({
+				version: "java" + majorVersion,
+				releaseTime: entries.at(-1)!.timestamp.toISOString(),
+
+				runtimes: entries.flatMap(transformRuntime)
+			});
+		}
+
+		return result;
+	},
+	recommend: () => false,
+});
+
+function isAvailableBinary(binary: AdoptiumJavaBinary): boolean {
+	if (binary.os !== "linux" && binary.os !== "windows" && binary.os !== "mac") {
+		return false;
+	}
+
+	if (binary.architecture !== "x64" && binary.architecture !== "x86" && binary.architecture !== "aarch64" && binary.architecture !== "arm") {
+		return false;
+	}
+
+	return true;
+}
+
+function getOSType(binary: AdoptiumJavaBinary): string {
+	let osName = binary.os;
+	if (osName === "mac") {
+		osName = 'mac-os';
```

Comment:
Missed some!
In future it would be nice if you opened a PR from a fork on your own account as that allows maintainers to fix things. Also it is possible to apply suggestions in bulk which you didn't seem to realise
```suggestion
		osName = "mac-os";
```